### ISSUE TITLE
Updating a mod to a new minecraft version only works if the mod version has also changed

### ISSUE THREAD

**Issue by TayouVR** [PrismLauncher/PrismLauncher#2687]
### Operating System

Windows

### Version of Prism Launcher

9.0 develop

### Version of Qt

6.7.1

### Description of bug

When updating a instances minecraft version I use the mod updater to find mods compatible with the new version.
However 3D Skin Layers didn't bump its own version, it just released a new version for 1.21 with the same mod version number. Therefore the launcher doesn't update it:
![image](https://github.com/user-attachments/assets/cedf23a1-2bd8-4c4d-ab26-f5cd505cadbf)
![image](https://github.com/user-attachments/assets/02ffb8ea-01c2-466b-8144-346f1f98e50a)

When downloading the mod directly it is disabled by default in the confirmation prompt:
![image](https://github.com/user-attachments/assets/2c8ca712-acef-4b0e-bf7c-835297fe997d)
![image](https://github.com/user-attachments/assets/1d88416f-4a99-4587-8ce0-d38bfb661d6c)

### Steps to reproduce

- create a instance with minecraft 1.20.4, fabric or quilt
- download 3D skin layers, newest version (1.6.6)
- change minecraft version of the instance to 1.21
- observe above described behaviour on mod update

### Suspected cause

game version number is not taken into consideration when checking for mod updates since the game version info metadata has only recently been added.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Trial97** [comment]
I thought that was obvious: if the versions are the same we do not need to redownload the mod.
But your use case may not be yet supported: we never planned to use the update mods to update them for the new Minecraft version(what we have now works due to that version increment)

**Comment by TayouVR** [comment]
I think for the end user updating to the same version, but for the minecraft version of their current instance would be the logical action. At least I think thats generally what people want to happen.

**Comment by TheKodeToad** [comment]
Is this not the mods' issue? A version number should always correspond to the same file content.

**Comment by TayouVR** [comment]
> Is this not the mods' issue? A version number should always correspond to the same file content.

no, mods quite often release a version with like 7 files for different minecraft versions. They all have the same version number.

The version number only gets bumped with code changes in the mod. So if none are required for adding a MC version or if just making a new release this totally makes sense. Look at it like our release artifacts/CI builds.
we don't bump the version for every platform either.

**Comment by Rozbiynk** [comment]
> I thought that was obvious: if the versions are the same we do not need to redownload the mod. But your use case may not be yet supported: we never planned to use the update mods to update them for the new Minecraft version(what we have now works due to that version increment)

can you add check for mc versions, instead of only file version?

**Comment by Trial97** [comment]
@TayouVR this looks like a duplicate of https://github.com/PrismLauncher/PrismLauncher/issues/2134 
Can we close one of them?

**Comment by TayouVR** [comment]
they are similar issues, but not the same. A dedicated button to switch mods to the correct minecraft version could also make sense.
But this issue is about the mod update button.

**Comment by Bodrew** [comment]
+1, would like to request the check for updates feature be enhanced beyond just file name/mod version. With Mojang going from 1.21.6 to 1.21.8 in the span of just a month, I assume mod developers have stopped incrementing their mod versions just to keep up with MC versioning, but are in fact pushing updates. 

Sorry if this is a necro.

**Comment by hannibal002** [comment]
+1

**Comment by Sakth1** [comment]
+1. This would be a lovely addition.
